### PR TITLE
Adds Result lists and handling

### PR DESCRIPTION
**Pull request by ZephyrTFA**
## About The Pull Request

Useful boilerplate for error handling.
Inspired by rust.

Implemented it for GAGs in coordination with Ninja.

### CODE REVIEW

File: code/__DEFINES/results.dm [hunk 1]
```diff
@@ -0,0 +1,31 @@
+/// The return type which is a specially handled list which can contain both an error and the expected result.
+/// This define is so you can at a glance know if something is just returning a normal list or a Result list
+#define RESULT /list
```
Comment (LemonInTheDark) [inline code/__DEFINES/results.dm:3]: out of curiosity, why macro wrap it instead of literally just doing a datum?

**Comment by LemonInTheDark** [comment]
do we actually want to bubble up error messages here? we don't have types for errors so we can't react differently to them, and there isn't much of a reason to handle them distinctly at least that immediately comes to mind.
what's the general sell here?

File: code/game/machinery/announcement_system.dm [hunk 2]
```diff
@@ -172,7 +172,7 @@ GLOBAL_LIST_EMPTY(announcement_systems)
 		return
 
 	arrival = pick("#!@%ERR-34%2 CANNOT LOCAT@# JO# F*LE!", "CRITICAL ERROR 99.", "ERR)#: DA#AB@#E NOT F(*ND!")
-	newhead = pick("OV#RL()D: \[UNKNOWN??\] DET*#CT)D!", "ER)#R - B*@ TEXT F*O(ND!", "AAS.exe is not responding. NanoOS is searching for a solution to the problem.")
+	newhead = pick("OV#RL()D: \[UNKNOWN??] DET*#CT)D!", "ER)#R - B*@ TEXT F*O(ND!", "AAS.exe is not responding. NanoOS is searching for a solution to the problem.")
```
Comment (LemonInTheDark) [inline code/game/machinery/announcement_system.dm:175]: this looks like a mistake

Comment (ZephyrTFA) [reply code/game/machinery/announcement_system.dm:175]: It's not, \] is not required

Comment (LemonInTheDark) [reply code/game/machinery/announcement_system.dm:175]: does it hurt anything? the consistency seems nice. also like why this pr

**Pull request closed without merging**